Packs/ThreatGrid/Integrations/ThreatGrid/ThreatGrid.py

```diff
@@ -1,1233 +1,1520 @@
+            artifact (str, optional): The artifact to download.
+                Sample ID is required when choosing 'artifact'.
+
+        Returns:
+            Dict[str, Any]: API response from Cisco ThreatGrid.
+        """
+        params = remove_empty_elements({
+            "limit": limit,
+            "offset": offset,
+        })
 
+        url_suffix = f"samples/{sample_id}" if sample_id else "samples"
+        url_suffix = f"{url_suffix}/summary" if summary else url_suffix
+
+        if artifact:
+            resp_type = "text"
+            url_suffix = f"{url_suffix}/{artifact}"
+        else:
+            resp_type = "json"
+
+        return self._http_request("GET",
+                                  f"{API_V2_PREFIX}{url_suffix}",
+                                  params=params,
+                                  resp_type=resp_type)
+
+    def analysis_sample_request(
+        self,
+        sample_id: str,
+        analysis_type: str,
+        arg_value: Optional[str] = None,
+    ) -> Dict[str, Any]:
+        """Get analysis data for a specific sample.
+
+        Args:
+            sample_id (str): The sample ID.
+            analysis_type (str): URL parameter (processes/network-streams/iocs/annotations).
+            arg_value (str, optional): argument value for the URL parameter
+                    (specific process/network-stream/ioc).
+
+        Returns:
+            Dict[str, Any]: API response from Cisco ThreatGrid.
+        """
+        analysis_type = f'{analysis_type}s' if analysis_type == 'network_stream' else analysis_type
+        url_prefix = f"{analysis_type}/{arg_value}" if arg_value else analysis_type
+
+        return self._http_request("GET",
+                                  f"{API_V2_PREFIX}samples/{sample_id}/analysis/{url_prefix}")
+
+    def whoami_request(self) -> Dict[str, Any]:
+        """Get details about correct login user and organization.
+
+        Returns:
+            Dict[str, Any]: API response from Cisco ThreatGrid.
+        """
+        return self._http_request("GET", f"{API_V3_PREFIX}session/whoami")
+
+    def list_associated_samples_request(
+        self,
+        arg_name: str,
+        arg_value: str,
+        limit: Optional[int] = None,
+        offset: Optional[int] = None,
+    ) -> Dict[str, Any]:
+        """Returns a list of samples associated to the domain /
+            IP / URL / path / artifact / registry key that specified.
+
+        Args:
+            arg_name (str): argument name (URL parameter).
+            arg_value (str): argument value (specific value for the URL parameter).
+
+        Returns:
+            Dict[str, Any]: API response from Cisco ThreatGrid.
+        """
+        params = remove_empty_elements({
+            "limit": limit,
+            "offset": offset,
+        })
+        return self._http_request("GET",
+                                  f"{API_V2_PREFIX}{arg_name}s/{arg_value}/samples",
```

Comment:
please use urljoin when joining API_V2_PREFIX and anything else. It prevents bugs like double slashes, or missing slashes.